lms/djangoapps/student_account/views.py

```diff
@@ -302,6 +306,69 @@ def _external_auth_intercept(request, mode):
         return external_auth_register(request)
 
 
+def get_user_orders(user):
+    """Given a user, get the detail of all the orders from the Ecommerce service.
+
+    Arguments:
+        user (User): The user to authenticate as when requesting ecommerce.
+
+    Returns:
+        list of dict, representing orders returned by the Ecommerce service.
+    """
+    no_data = []
+    commerce_configuration = CommerceConfiguration.current()
+    if not commerce_configuration.enabled:
+        log.warning('%s configuration is disabled.', commerce_configuration.API_NAME)
+        return no_data
+
+    if commerce_configuration.is_cache_enabled:
+        cache_key = commerce_configuration.CACHE_KEY + '.' + user.username
+        cached = cache.get(cache_key)
+        if cached:
+            return cached
+
+    user_orders = []
+    allowed_course_modes = ['professional', 'verified', 'credit']
+    try:
+        response = ecommerce_api_client(user).orders.get()
+        commerce_user_orders = response.get('results', no_data)
+        page = 1
+        next_page = response.get('next', None)
+        while next_page:
+            page += 1
+            response = ecommerce_api_client(user).orders.get({'page': page})
+            commerce_user_orders += response.get('results', no_data)
+            next_page = response.get('next', None)
+    except Exception:  # pylint: disable=broad-except
+        log.exception('Failed to retrieve data from the Commerce API.')
+        return no_data
+
+    for order in commerce_user_orders:
+        if order['status'].lower() == 'complete':
+            for attribute in order['lines'][0]['product']['attribute_values']:
```

Comment:
Why only do this for the first line in the order? What about a multi-line order? Otto supports those.